libmproxy/protocol/primitives.py

```diff
@@ -236,7 +236,7 @@ def change_server(
         ssl_mismatch = (
             ssl is not None and
             (
-                ssl != self.c.server_conn.ssl_established
+                (self.c.server_conn.connection and ssl != self.c.server_conn.ssl_established)
```

Comment:
TL;DR this ensures that attempts to upgrade a server connection to SSL will not establish a new connection when one didn't exist before.

When ssl_mismatch is True, this code always tries to connect to the upstream server (in an attempt to correct the mismatch). My other code changes ensure that server_conn.ssl_established is always False because it hasn't connected to the server. That would mean that without this change to ssl_mismatch, it would still try to create a new connection to the server.

If I understand the rest of this method correctly, my change will only affect the case where the server is the same, it is attempting an upgrade to SSL, and doesn't already have a server connection.